version: "3.8"

services:
  glitchandbooks:
    build: .
    container_name: glitchandbooks
    env_file:
      - .env
    depends_on:
      - db-glitchandbooks
    networks:
      - traefik
    restart: always
    ports:
      - "5000:5000"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.glitchandbooks.rule=Host(`glitchandbooks.fr`)"
      - "traefik.http.routers.glitchandbooks.entrypoints=websecure"
      - "traefik.http.routers.glitchandbooks.tls=true"
      - "traefik.http.services.glitchandbooks.loadbalancer.server.port=5000"
      - "traefik.http.routers.glitchandbooks.tls.certresolver=myresolver"
      - "traefik.docker.network=traefik"

  db-glitchandbooks:
    image: postgres:13
    container_name: db-glitchandbooks
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASS}
      POSTGRES_DB: ${DB_NAME}
    volumes:
      - pgdata-glitchandbooks:/var/lib/postgresql/data
    ports:
      - "5433:5432"
    networks:
      - traefik
    restart: always

volumes:
  pgdata-glitchandbooks:

networks:
  traefik:
    external: true
